pandas/core/series.py

```diff
@@ -1122,6 +1122,248 @@ def _set_value(self, label, value, takeable: bool = False):
+        in combination with the source code in :mod:`pandas.core.computation.parsing`.
+
+        Examples
+        --------
+        >>> series = pd.Series(range(5))
+        >>> series
+        0    0
+        1    1
+        2    2
+        3    3
+        4    4
+        dtype: int64
+        >>> series.query('index > 2')
+        3    3
+        4    4
+        dtype: int64
+
+        If the index is named, we can refer to it in the query.
+
+        >>> series.index.name = 'numbers'
+        >>> series.query('numbers > 2')
+        numbers
+        3    3
+        4    4
+        dtype: int64
+        """
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        if not isinstance(expr, str):
+            msg = f"expr must be a string to be evaluated, {type(expr)} given"
+            raise ValueError(msg)
+        kwargs["level"] = kwargs.pop("level", 0) + 1
+        kwargs["target"] = None
+        res = self.eval(expr, **kwargs)
+
+        try:
+            new_data = self.loc[res]
+        except ValueError:
+            # when res is multi-dimensional loc raises, but this is sometimes a
+            # valid query
+            new_data = self[res]
+
+        if inplace:
+            self._update_inplace(new_data)
+        else:
+            return new_data
+
+    def eval(self, expr, inplace=False, **kwargs):
+        """
+        Evaluate a string describing operations on Series columns.
+
+        Operates on columns only, not specific rows or elements.  This allows
+        `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to evaluate.
+        inplace : bool, default False
+            If the expression contains an assignment, whether to perform the
+            operation inplace and mutate the existing Series. Otherwise,
+            a new Series is returned.
+        **kwargs
+            See the documentation for :func:`eval` for complete details
+            on the keyword arguments accepted by
+            :meth:`~pandas.Series.query`.
+
+        Returns
+        -------
+        ndarray, scalar, or pandas object
+            The result of the evaluation.
+
+        See Also
+        --------
+        Series.query : Evaluates a boolean expression to query the columns
+            of a frame.
+        eval : Evaluate a Python expression as a string using various
+            backends.
+
+        Notes
+        -----
+        For more details see the API documentation for :func:`~eval`.
+        For detailed examples see :ref:`enhancing performance with eval
+        <enhancingperf.eval>`.
+
+        Examples
+        --------
+        >>> series = pd.Series(1, index=range(5))
+        >>> series
+        0    1
+        1    1
+        2    1
+        3    1
+        4    1
+        dtype: int64
```

Comment:
rather than copying the doc-string, we need to share the same one with DataFrame. we do this with lots of methods, e.g. the impl goes in generic.py and then we have a stub in series.py and frame.py

Reply:
you can get fancy and add examples separately but to be honest its not worth it. simply add a single series example in the current docs.